responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -50,10 +51,12 @@ class RAIInsights(object):
     """
 
     def __init__(self, model, train, test, target_column,
-                 task_type, categorical_features=None, classes=None,
+                 task_type, categorical_features=None,
+                 metadata_columns=None, classes=None,
```

Comment:
Can you add this parameter in the end after maximum_rows_for_test for back compat reasons?

Reply:
Logically, I felt like it makes most sense right after `categorical_features`, since it's also a list of column names. Realizing that these are positional arguments actually makes me want to make all but model, train, and test keyword-only:

```
func(self, model, train, test, *, target_column, task_type, categorical_features=None, ...)
```
In that case, the positioning after `*` is irrelevant since you have to call it with keyword
```
obj.func(model, train, test, target_column="y", categorical_features=["gender"], metadata_columns=["gender", "age"], task_type="classification", ...)
``` 

To me, it's pretty clear that there's no benefit at all to having them be positional. We've done something like this in Fairlearn a while ago: https://github.com/fairlearn/fairlearn/blob/862263f4352e23da088ae7a638b88b56580e5230/fairlearn/metrics/_metric_frame.py#L269
Since it's not a new class, but one that already exists, the situation is somewhat more complicated. Making any change might break code that would currently run fine. If I remember correctly, we ended up adding a warning that arguments will be keyword-only starting in the next released version and then made the change one version later. For this particular PR I won't get around moving it to the end it seems.

@xuke444 @riedgar-ms @gaugup @imatiach-msft

Reply:
we only just very recently released this RAIInsights, so I think it's fine to change as long as we have a really really really good reason to.  I'm not quite sure if this one meets the bar.